pkg/config/validation/validation.go

```diff
@@ -1024,46 +1019,18 @@ func ValidateProxyAddress(hostAddr string) error {
 	return nil
 }
 
-// ValidateDurationGogo checks that a gogo proto duration is well-formed
-func ValidateDurationGogo(pd *types.Duration) error {
-	dur, err := types.DurationFromProto(pd)
-	if err != nil {
-		return err
-	}
-	if dur < time.Millisecond {
-		return errors.New("duration must be greater than 1ms")
-	}
-	if dur%time.Millisecond != 0 {
-		return errors.New("only durations to ms precision are supported")
-	}
-	return nil
-}
-
 // ValidateDuration checks that a proto duration is well-formed
 func ValidateDuration(pd *types.Duration) error {
-	dur, err := types.DurationFromProto(pd)
-	if err != nil {
-		return err
-	}
-	if dur < time.Millisecond {
+	duration := pd.AsDuration()
+	if time.Duration.Milliseconds(duration) > 1 {
 		return errors.New("duration must be greater than 1ms")
 	}
-	if dur%time.Millisecond != 0 {
```

Comment:
I think we are probably loosing this check?

Reply:
time.Duration stores an int64 in nanoseconds: https://golang.org/src/time/time.go?s=20758:20777#L573 although my check on 1025 has the wrong conditional

Reply:
The problem is envoy does not - so if you specify some number of nanosecond precision it won't work

Reply:
I see, thanks for the clarity.